contrib/packs/actions/pack_mgmt/download.py

```diff
@@ -48,11 +48,47 @@ class DownloadGitRepoAction(Action):
     def __init__(self, config=None, action_service=None):
         super(DownloadGitRepoAction, self).__init__(config=config, action_service=action_service)
 
+        self.https_proxy = os.environ.get('https_proxy', self.config.get('https_proxy', None))
+        self.http_proxy = os.environ.get('http_proxy', self.config.get('http_proxy', None))
+        self.proxy_ca_bundle_path = os.environ.get(
+            'proxy_ca_bundle_path',
+            self.config.get('proxy_ca_bundle_path', None)
+        )
+        self.no_proxy = os.environ.get('no_proxy', self.config.get('no_proxy', None))
+
+        self.proxy_config = None
+
+        if self.http_proxy or self.https_proxy:
+            self.logger.debug('Using proxy %s',
+                              self.http_proxy if self.http_proxy else self.https_proxy)
+            self.proxy_config = {
+                'https_proxy': self.https_proxy,
+                'http_proxy': self.http_proxy,
+                'proxy_ca_bundle_path': self.proxy_ca_bundle_path,
+                'no_proxy': self.no_proxy
+            }
+
+        # This is needed for git binary to work with a proxy
+        if self.https_proxy and not os.environ.get('https_proxy', None):
+            os.environ['https_proxy'] = self.https_proxy
+
+        if self.http_proxy and not os.environ.get('http_proxy', None):
+            os.environ['http_proxy'] = self.http_proxy
+
+        if self.no_proxy and not os.environ.get('no_proxy', None):
+            os.environ['no_proxy'] = self.no_proxy
+
+        if self.proxy_ca_bundle_path and not os.environ.get('proxy_ca_bundle_path', None):
+            os.environ['no_proxy'] = self.no_proxy
+
+        if self.http_proxy and self.https_proxy:
+            raise Exception('Either specify "http_proxy" or "https_proxy" but not both.')
```

Comment:
@armab @Kami curl accepts both arguments at the same time and based on the protocol picks the right proxy. pip however picks one proxy argument and makes the assumption that you'll use the proxy for both http/https proxy. https://github.com/pypa/pip/blob/master/pip/basecommand.py#L94. Things like squid doesn't allow http and https on same port. Things like charles proxy however do allow both protocols to work on same port. So it seems like the "right" behavior is to allow both the variables to be specified. If those variables point to different proxies for http and https, pip won't work but it seems more like a pip issue. So I am going to revert the last commit to throw exception on specifying both variables.